python/pylibvw.cc

```diff
@@ -857,6 +873,50 @@ uint32_t ex_get_cbandits_class(example_ptr ec, uint32_t i) { return ec->l.cb.cos
 float ex_get_cbandits_probability(example_ptr ec, uint32_t i) { return ec->l.cb.costs[i].probability; }
 float ex_get_cbandits_partial_prediction(example_ptr ec, uint32_t i) { return ec->l.cb.costs[i].partial_prediction; }
 
+size_t ex_get_ccb_type(example_ptr ec)
+{
+  switch(ec->l.conditional_contextual_bandit.type)
+  {
+    case CCB::example_type::shared:
+      return tSHARED;
+    case CCB::example_type::action:
+      return tACTION;
+    case CCB::example_type::slot:
+      return tSLOT;
+    default:
+      return tUNSET;
+  }
+}
+py::object* ex_get_ccb_cost(example_ptr ec)
+{
+  auto* outcome_ptr = ec->l.conditional_contextual_bandit.conditional_contextual_bandit_outcome;
+  return outcome_ptr == nullptr ? Py_None : value_to_pyobject<float>(outcome_ptr->cost);
+}
+uint32_t ex_get_ccb_num_included_actions(example_ptr ec)
+{
+  const auto& label = ec->l.conditional_contextual_bandit;
+  return label.explicit_included_actions.size();
+}
+py::object* to_pyobject ex_get_ccb_class(example_ptr ec, uint32_t i)
+{
+  if (i >= ex_get_ccb_num_included_actions(ec)) { THROW("Class index out of bounds"); }
+  auto* outcome_ptr = ec->l.conditional_contextual_bandit.conditional_contextual_bandit_outcome;
+  return outcome_ptr == nullptr ? Py_None : value_to_pyobject<uint32_t>(outcome_ptr->probabilities[i]->action);
```

Comment:
@jackgerrits I suppose this is the correct way?

Reply:
Looks like object could be used for this?
https://www.boost.org/doc/libs/1_64_0/libs/python/doc/html/tutorial/tutorial/object.html

Not sure if this will work but a look at that file makes me think it should work this way:
```c++
py::object to_pyobject ex_get_ccb_class(example_ptr ec, uint32_t i)
{
  if (i >= ex_get_ccb_num_included_actions(ec)) { THROW("Class index out of bounds"); }
  auto* outcome_ptr = ec->l.conditional_contextual_bandit.conditional_contextual_bandit_outcome;
  return outcome_ptr == nullptr ? py::object() : py::object(outcome_ptr->probabilities[i]->action);
}
```